packages/core/src/util/slotCalc.ts

```diff
@@ -119,30 +120,50 @@ export const createSlotEpochInfoCalc = (eraSummaries: EraSummary[]) => {
 };
 
 /**
- * @returns first slot of the epoch
+ * The actual implementation, _memoized_ by `epochSlotsCalc` or directly used by `epochSlotsCalcFactory`.
+ *
+ * @param epochNo epoch number
+ * @param eraSummaries era summaries array
+ * @returns `eraSummary`, `firstSlot` and `lastSlot` of given epoch
  */
-export const epochSlotsCalc = memoize((epochNo: EpochNo, eraSummaries: EraSummary[]) => {
+const epochSlotsCalcImplementation = (epochNo: EpochNo, eraSummaries: EraSummary[]) => {
   let atEpoch = 0;
   let atSlot = eraSummaries[0].start.slot;
   let eraSummaryIdx = 0;
   const maxEraSummaryIdx = eraSummaries.length - 1;
+
   while (atEpoch !== epochNo) {
     atSlot += eraSummaries[eraSummaryIdx].parameters.epochLength;
     atEpoch++;
+
     if (eraSummaryIdx < maxEraSummaryIdx && atSlot >= eraSummaries[eraSummaryIdx + 1].start.slot) {
       eraSummaryIdx++;
     }
   }
+
+  const eraSummary = eraSummaries[eraSummaryIdx];
+
   return {
+    eraSummary,
     firstSlot: Slot(atSlot),
-    lastSlot: Slot(atSlot + eraSummaries[eraSummaryIdx].parameters.epochLength - 1)
+    lastSlot: Slot(atSlot + eraSummary.parameters.epochLength - 1)
   };
-});
+};
 
 /**
- * @returns first and last slot numbers of the epoch
+ * @returns `eraSummary`, `firstSlot` and `lastSlot` of given epoch
  */
-export type EpochSlots = ReturnType<typeof epochSlotsCalc>;
+export const epochSlotsCalc = memoize(epochSlotsCalcImplementation);
```

Comment:
This is a breaking change, please reflect in the commit message

Reply:
db4d771dd should solve this

Reply:
db4d771dd should solve this

Reply:
Same as https://github.com/input-output-hk/cardano-js-sdk/pull/882#discussion_r1371182775

Reply:
2a19704cc should solve this